### ISSUE TITLE
`&amp;` parsing wrong when `xmlns` points to XHTML

### ISSUE THREAD

**Issue by weiwu-zhang** [xmldom/xmldom#203]
````
$ cat xmldom_bug_22.js 
const { DOMParser } = require('@xmldom/xmldom')

const items = [
  `<html xmlns="http://www.w3.org/1999/xhtml"><script>let message = " &amp; ETH";</script></html>`,
  `<html><script>let message = " &amp; ETH";</script></html>`,
];

for (const item of items) {
  const xml = new DOMParser().parseFromString(item, "application/xml");
  console.log(xml.documentElement.childNodes[0].textContent);
}
$ node xmldom_bug_22.js 
let message = " &amp; ETH";
let message = " & ETH";
````

Where both messages should be the same. Something trigged the misintrepreation when `xmlns="http://www.w3.org/1999/xhtml"` presents.

P.S. this is not a duplicate of #58

**Comment by karfau** [comment]
@colourful-land Thx for reporting this, providing a test scenario and linking all related issues.

1. Just to be sure: What versions of xmldom have you tried this with?

2. Did I understand correctly, that you expect the output to be `let message = " & ETH";` in both cases?

**Comment by colourful-land** [comment]
>     1. Just to be sure: What versions of xmldom have you tried this with?

I first tried 0.5.0 - failing the test, I tried again with the master branch from github, still fails.

>     2. Did I understand correctly, that you expect the output to be `let message = " & ETH";` in both cases?

Yes, as it is the case if the code runs in a browser using the browser's dom parser.

**Comment by karfau** [comment]
FYI: I have a rough plan how to fix this issue (I already invested quite some time into a draft so far), but it's not a trivial one, and it's a breaking change.

The main issue is that xmldom currently "detects" the default HTML namespace when parsing and applies certain rules, that should **only** be applied when setting the second argument `mimeType` of `DOMParser.parseFromString()` to `'text/html'`. (According to [the most current spec](https://html.spec.whatwg.org/multipage/dynamic-markup-insertion.html#dom-domparser-parsefromstring).)

Since there will be quite some changes and quite some missing tests for the areas of code that I need to touch, I will file multiple PRs before we can resolve this issue.

**Comment by colourful-land** [comment]
Thank you very much @karfau. Right now I simply replace all occurrences of xhtml namespace in lib/sax.js and lib/dom.js to keep my project going. Hope to see a new release so I can rever to use unmodified xmldom in the near future!

**Comment by karfau** [comment]
@colourful-land Interesting idea. I hope it helps more than it does any damage. Maybe you run the tests to see what things change/break. Just to be sure this doesn't introduce any serious bugs.

Let m know in case any support is needed.

I'm also looking forward to the next release(s), but can not promise any timeline. :smile:

**Comment by karfau** [comment]
I needed to shift this into the next breaking release. I already invested quite some time into it, but it's not done and I don't want to hold back 0.8.0 any longer, which already has other potentially breaking changes (and mixing up to many of them is also not so nice for people doing updates).

**Comment by karfau** [comment]
Great news: The version I have in #338 no longer has the issue you describe.

It still is quite a massive change, so it will still need some time to get the PR done.

With all the breaking changes I think I will try to create a pre-release, so you (and others can test it before we release it as 0.9.0

**Comment by karfau** [comment]
@weiwu-zhang I finally managed to merge and release the PR before it managed to be alive for a whole year.
I created a pre-releaseas mentioned, so it will only be installable as `@xmldom/xmldom@next` or `@xmldom/xmldom@0.9.0-beta.1`.

THank you fro your endurance and let me know in case you find any issues.
